etna/collections/models.py

```diff
@@ -70,6 +73,9 @@ class TopicExplorerIndexPage(TeaserImageMixin, MetadataPageMixin, BasePage):
     ]
     promote_panels = MetadataPageMixin.promote_panels + TeaserImageMixin.promote_panels
 
+    # DataLayerMixin overrides
+    gtm_content_group = "Explorer"
```

Comment:
How about `gtm_content_group = ExplorerIndexPage.gtm_content_group` ?

Reply:
Because it's just a string. There's no reason why any of the strings on any of the models shouldn't be able to change at any point in time, and that shouldn't impact any other page type.

Reply:
In other words... its purely coincidence that the value is the same for all page types in this module. There's absolutely no need for that coincidence to influence the implementation.

Reply:
Yes, but what about when the value changes for one model, but not another?

Reply:
> Yes, but what about when the value changes for one model, but not another?

Since the `gtm_content_group`  can override  DataLayerMixin, it can be updated to the required values when required, until then it adopts the common value.

Reply:
Sorry Allan, but the whole idea behind this refactor is to simplify this as much as possible. I know it's very natural to want to reduce duplication in the codebase, but I really don't think it's worthwhile in this particular case. It should be clear from a ticket's requirements what needs to change and what doesn't, and a mass search/replace is just as easy to action as single one.

Reply:
>... but the whole idea behind this refactor is to simplify this as much as possible. I know it's very natural to want to reduce duplication in the codebase, but I really don't think it's worthwhile in this particular case. 

The existing implementation for `get_gtm_content_group`  avoided duplication of code as it done now. 


> _Yes, but what about when the value changes for one model, but not another?_

_Since the gtm_content_group can override DataLayerMixin, it can be updated to the required values when required, until then it adopts the common value._

What is the problem with accessing another class attribute? 

> It should be clear from a ticket's requirements what needs to change and what doesn't, 

Do you want to clarify the requirements and proceed from there?

> a mass search/replace is just as easy to action as single one.

This does not look good. That is why we have `etna\ciim\constants.py`

Reply:
I'm sorry Allan, but this is not what constants are for. They are for variables that have wider meaning throughout the codebase. These are just 'labels' without any wider meaning. You look at the class, you know what the value is just by looking at the code, you change it if it needs to change, and it doesn't affect anything else... Pure and simple.

Reply:
> You look at the class, you know what the value is just by looking at the code, you change it if it needs to change, and it doesn't affect anything else... Pure and simple.

This seems contradicting to what you mentioned. -
>  and a mass search/replace is just as easy to action as single one.

Previous queries:
> What is the problem with accessing another class attribute?

Is there an issue with this approach?

> Do you want to clarify the requirements and proceed from there?

Reply:
> This seems contradicting to what you mentioned

I'm honestly not seeing what you mean. Both my points prioritise simplicity and defend the current design. Using constants or some reference to a value defined elsewhere... It's all just unnecessary complexity. Like I said at the beginning: It is only coincidence that models share this value. There's no reason the value for any model shouldn't be changed in future.

Reply:
> Using constants or some reference to a value defined elsewhere... It's all just unnecessary complexity.

I thought using a reference to a value defined elsewhere will eliminate the trouble of updating the same value everywhere.  That's why I said
> The same string name is used across different pages. Making changes in one place is convenient than same changes in various places.

> It's all just unnecessary complexity.

If your view is about making some reference to a value defined elsewhere in this case _complex_, I will go with your experienced view to keep the same name defined for `gtm_content_group` across different models.